fix: Alternate reschedule behaviour for past bookings - Don't allow new booking

hariombalhara opened the pull request:
## What does this PR do?

This PR provides an alternative default behaviour for past bookings reschedule. By default, past bookings if rescheduled, allow the user to create a new booking instead. Now, if __CAL_INTERNAL_PAST_BOOKING_RESCHEDULE_CHANGE_ has the teamId for the event then the default behaviour would be to redirect to booking details/status page. This would make sure that no booking can be done from past booking's reschedule link.

Fixes PRI-311

## Key Changes
- **Environment variable support**: Implemented `ENV_PAST_BOOKING_RESCHEDULE_CHANGE` to control team-specific behavior for past booking reschedules
- **Comprehensive test coverage**: Added extensive unit tests to validate the new redirect behavior

## Visual Demo (For contributors especially)


## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

### Environment Variables
- Set `_CAL_INTERNAL_PAST_BOOKING_RESCHEDULE_CHANGE` environment variable with comma-separated team IDs to test the new behavior

### Test Scenarios



## Checklist

- I have read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code follows the style guidelines of this project
- I have commented my code, particularly in hard-to-understand areas
- I have checked if my changes generate no new warnings

hariombalhara committed:
fix: Past booking redirect

hariombalhara committed:
feat: past booking redirect - alternate behaviour

anikdhabal commented on packages/lib/constants.ts:
```diff
@@ -249,3 +249,7 @@ export const RETELL_AI_TEST_EVENT_TYPE_MAP = (() => {
+
+// Environment variable for configuring past booking reschedule behavior per team. A comma separated list of team IDs(e.g. '1,2,3')
+/* This is an internal environment variable and is not meant to be used by the self-hosters. It is planned to be removed later by either having it as an option in Event Type or by some other customer configurable approaches*/
+export const ENV_PAST_BOOKING_RESCHEDULE_CHANGE = process.env._CAL_INTERNAL_PAST_BOOKING_RESCHEDULE_CHANGE;
```
When I first saw this, from the naming of the env it looked like a boolean type, but it’s actually meant to store team IDs. So NIT: maybe we should change the env name

anikdhabal replied:
Mybe something like:- `ENV_PAST_BOOKING_RESCHEDULE_CHANGE_TEAM_IDS` or `ENV_PAST_BOOKING_RESCHEDULE_CHANGE_ALLOWED_TEAMS`

devin-ai-integration[bot] committed:
fix: rename ENV_PAST_BOOKING_RESCHEDULE_CHANGE to ENV_PAST_BOOKING_RESCHEDULE_CHANGE_TEAM_IDS for clarity

- Renamed environment variable to better reflect that it contains comma-separated team IDs
- Updated all references in constants.ts, implementation file, tests, and turbo.json
- Addresses PR comment feedback about misleading variable name

Co-Authored-By: hariom@cal.com <hariombalhara@gmail.com>

anikdhabal reviewed (approved):
Looks good

Udit-takkar committed:
Merge branch 'main' into 08-27-fix_past_booking_redirect
